### PR TITLE
added limit param for order_book with json default of 20

### PR DESCRIPTION
**Pull request by nikhilsaraf**
closes https://github.com/stellar/horizon/issues/399

### CODE REVIEW

**Comment by nikhilsaraf** [comment]
screenshot with default limit (shows 2 of 2 orders; default is limit = 20):
screen shot 2017-10-11 at 4 52 48 pm

screenshot with default limit = 1 (shows 1 of 2 orders):
screen shot 2017-10-11 at 4 52 20 pm

screenshot with default limit = 5 (shows 2 of 2 orders):
screen shot 2017-10-11 at 4 52 34 pm

**Comment by nikhilsaraf** [comment]
only concern is backward compatibility with existing Go SDK users, who will now need to specify a limit param for calls to LoadOrderBook (clients/horizon/client.go) -- not sure what our policy is for such things.

besides this, I don't have any more planned changes

**Comment by nikhilsaraf** [comment]
added a test fix caused by the backward compatibility, should be good to go if the tests pass now.

File: services/horizon/docs/reference/endpoints/orderbook-details.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Horizon will return, for each orderbook, a summary of the orderbook and the bids
 ## Request
 
 ```
-GET /order_book?selling_asset_type={selling_asset_type}&selling_asset_code={selling_asset_code}&selling_asset_issuer={selling_asset_issuer}&buying_asset_type={buying_asset_type}&buying_asset_code={buying_asset_code}&buying_asset_issuer={buying_asset_issuer}
+GET /order_book?selling_asset_type={selling_asset_type}&selling_asset_code={selling_asset_code}&selling_asset_issuer={selling_asset_issuer}&buying_asset_type={buying_asset_type}&buying_asset_code={buying_asset_code}&buying_asset_issuer={buying_asset_issuer}&limit={limit}
 ```
 
 ### Arguments
```
Comment (bartekn) [inline services/horizon/docs/reference/endpoints/orderbook-details.md]: `limit` description is missing here.

File: clients/horizon/client.go [hunk 2]
```diff
@@ -114,7 +114,7 @@ func (c *Client) SequenceForAccount(
 }
 
 // LoadOrderBook loads order book for given selling and buying assets.
-func (c *Client) LoadOrderBook(selling Asset, buying Asset) (orderBook OrderBookSummary, err error) {
+func (c *Client) LoadOrderBook(selling Asset, buying Asset, limit uint64) (orderBook OrderBookSummary, err error) {
```
Comment (bartekn) [inline clients/horizon/client.go]: I think we should make it consistent with other methods here and use variadic parameter `params`. Check [`LoadAccountOffers`](https://github.com/stellar/go/blob/cbb1882bee9996781a65d92c4ed8eaec55dd7dbd/clients/horizon/client.go#L41-L60) method.
We have only one parameter here now but I think using variadic parameter still has it's advantages:
* upgrading the `stellar/go` to a new version will not require code changes,
* devs can just omit passing param to request default number of rows,
* if we decide to implement paging here in a future, we can easily add a new param (probably `cursor`?) without introducing a new parameter.

Comment (nikhilsaraf) [reply clients/horizon/client.go]: good point, solves the backward compatibility issue, updated.

File: services/horizon/docs/reference/endpoints/orderbook-details.md (hunk 1 above)
Comment (nikhilsaraf) [reply services/horizon/docs/reference/endpoints/orderbook-details.md]: fixed

**Comment by nikhilsaraf** [comment]
made changes from comments, added a test in clients/horizon/main_test.go so we now have one with a limit specified and one without.

**Comment by nikhilsaraf** [comment]
yeah, +1, we should do that, and keep it versioned too for easier upgrades.

**Pull request merged**